lib/services/android-plugin-build-service.ts

```diff
@@ -164,113 +169,157 @@ export class AndroidPluginBuildService implements IAndroidPluginBuildService {
 
-				updatedManifestContent = await this.updateManifestContent(androidManifestContent, defaultPackageName);
-			} else {
-				updatedManifestContent = this.createManifestContent(defaultPackageName);
-			}
+		this.$fs.copyFile(allGradleTemplateFiles, pluginTempDir);
+		this.addCompileDependencies(platformsAndroidDirPath, buildGradlePath);
+		const runtimeGradleVersions = await this.getRuntimeGradleVersions(projectDir);
+		this.replaceGradleVersion(pluginTempDir, runtimeGradleVersions.gradleVersion);
+		this.replaceGradleAndroidPluginVersion(buildGradlePath, runtimeGradleVersions.gradleAndroidPluginVersion);
+	}
 
-			// write the AndroidManifest in the temp-dir/plugin-name/src/main
-			const pathToNewAndroidManifest = path.join(newPluginMainSrcDir, MANIFEST_FILE_NAME);
-			try {
-				this.$fs.writeFile(pathToNewAndroidManifest, updatedManifestContent);
-			} catch (e) {
-				throw new Error(`Failed to write the updated AndroidManifest in the new location - ${pathToNewAndroidManifest}`);
-			}
+	private async getRuntimeGradleVersions(projectDir: string): Promise<IRuntimeGradleVersions> {
+		const registryData = await this.$npm.getRegistryPackageData(TNS_ANDROID_RUNTIME_NAME);
+		let runtimeGradleVersions: IRuntimeGradleVersions = null;
+		if (projectDir) {
+			const projectRuntimeVersion = this.$platformService.getCurrentPlatformVersion(
+				this.$devicePlatformsConstants.Android,
+				this.$projectDataService.getProjectData(projectDir));
+			runtimeGradleVersions = this.getGradleVersions(registryData.versions[projectRuntimeVersion]);
+			this.$logger.trace(`Got gradle versions ${JSON.stringify(runtimeGradleVersions)} from runtime v${projectRuntimeVersion}`);
+		}
 
-			// copy all android sourceset directories to the new temporary plugin dir
-			for (const dir of androidSourceSetDirectories) {
-				// get only the last subdirectory of the entire path string. e.g. 'res', 'java', etc.
-				const dirNameParts = dir.split(path.sep);
-				const dirName = dirNameParts[dirNameParts.length - 1];
+		if (!runtimeGradleVersions) {
+			const latestRuntimeVersion = registryData["dist-tags"].latest;
+			runtimeGradleVersions = this.getGradleVersions(registryData.versions[latestRuntimeVersion]);
+			this.$logger.trace(`Got gradle versions ${JSON.stringify(runtimeGradleVersions)} from the latest runtime v${latestRuntimeVersion}`);
+		}
 
-				const destination = path.join(newPluginMainSrcDir, dirName);
-				this.$fs.ensureDirectoryExists(destination);
+		return runtimeGradleVersions || {};
+	}
 
-				this.$fs.copyFile(path.join(dir, "*"), destination);
-			}
+	private getGradleVersions(packageData: { gradle: { version: string, android: string }}): IRuntimeGradleVersions {
+		const packageJsonGradle = packageData && packageData.gradle;
+		let runtimeVersions: IRuntimeGradleVersions = null;
+		if (packageJsonGradle && (packageJsonGradle.version || packageJsonGradle.android)) {
+			runtimeVersions = {};
```

Comment:
We can remove this line and initialize the variable when declaring it.

Reply:
Its here as we want to return null in such cases